login_server/src/cpp/SingletonManager/SessionManager.cpp

```diff
@@ -46,7 +46,7 @@ bool SessionManager::init()
 		case VALIDATE_NAME: mValidations[i] = new Poco::RegularExpression("^[^<>&;]{2,}$"); break;
 		case VALIDATE_USERNAME: mValidations[i] = new Poco::RegularExpression("^[a-zA-Z][a-zA-Z0-9_-]*$"); break;
 		case VALIDATE_EMAIL: mValidations[i] = new Poco::RegularExpression("^[a-zA-Z0-9.!#$%&?*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$"); break;
-		case VALIDATE_PASSWORD: mValidations[i] = new Poco::RegularExpression("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[@$!%*?&+-_])[A-Za-z0-9@$!%*?&+-_]{8,}$"); break;
+		case VALIDATE_PASSWORD: mValidations[i] = new Poco::RegularExpression("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^a-zA-Z0-9]).{8,}$"); break;
```

Comment:
Can we exclude the white space too here?
`"^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^a-zA-Z0-9 \t\n]).{8,}$"`

Reply:
And why doe we have two definitions?

Reply:
One is to check the whole password for all criteries at once and if this failed, it will be checked for all criterias separat until one critertia not met was found.

Reply:
We can exclude whitespace, tab and newline (and then also \r used in windows additional for newline and in classic mac)
but then we must add a new error message for this forbidden chars.
Else People are confused if the use one of this character and get only the error message that the password is invalid.

Reply:
OK. Then let us add another Test for the white space characters